pkg/controller/kafka/cluster/setup.go

```diff
@@ -186,3 +207,776 @@ func LateInitialize(cr *svcapitypes.ClusterParameters, obj *svcsdk.DescribeClust
+			}
+		} else if current.EncryptionAtRest != nil {
+			return false
+		}
+		if wanted.EncryptionInTransit != nil {
+			if current.EncryptionInTransit == nil {
+				return false
+			}
+			if aws.StringValue(wanted.EncryptionInTransit.ClientBroker) != aws.StringValue(current.EncryptionInTransit.ClientBroker) {
+				return false
+			}
+			if aws.BoolValue(wanted.EncryptionInTransit.InCluster) != aws.BoolValue(current.EncryptionInTransit.InCluster) {
+				return false
+			}
+		} else if current.EncryptionInTransit != nil {
+			return false
+		}
+	} else if current != nil {
+		return false
+	}
+	return true
+}
+
+func isClientAuthenticationUpToDate(wanted *svcapitypes.ClientAuthentication, current *svcsdk.ClientAuthentication) bool { // nolint:gocyclo
+	if wanted != nil {
+		if current == nil {
+			return false
+		}
+
+		if wanted.Sasl != nil {
+			if current.Sasl == nil {
+				return false
+			}
+			if wanted.Sasl.IAM != nil {
+				if current.Sasl.Iam == nil {
+					return false
+				}
+				if aws.BoolValue(wanted.Sasl.IAM.Enabled) != aws.BoolValue(current.Sasl.Iam.Enabled) {
+					return false
+				}
+			} else if current.Sasl.Iam != nil {
+				return false
+			}
+			if wanted.Sasl.Scram != nil {
+				if current.Sasl.Scram == nil {
+					return false
+				}
+				if aws.BoolValue(wanted.Sasl.Scram.Enabled) != aws.BoolValue(current.Sasl.Scram.Enabled) {
+					return false
+				}
+			} else if current.Sasl.Scram != nil {
+				return false
+			}
+		} else if current.Sasl != nil {
+			return false
+		}
+
+		if wanted.TLS != nil {
+			if current.Tls == nil {
+				return false
+			}
+			for _, wantedValue := range wanted.TLS.CertificateAuthorityARNList {
+				found := false
+				for _, currentValue := range current.Tls.CertificateAuthorityArnList {
+					if aws.StringValue(wantedValue) == aws.StringValue(currentValue) {
+						found = true
+						break
+					}
+				}
+				if !found {
+					return false
+				}
+			}
+			if aws.BoolValue(wanted.TLS.Enabled) != aws.BoolValue(current.Tls.Enabled) {
+				return false
+			}
+		} else if current.Tls != nil {
+			return false
+		}
+
+		if wanted.Unauthenticated != nil {
+			if current.Unauthenticated == nil {
+				return false
+			}
+			if aws.BoolValue(wanted.Unauthenticated.Enabled) != aws.BoolValue(current.Unauthenticated.Enabled) {
+				return false
+			}
+		} else if current.Unauthenticated != nil {
+			return false
+		}
+	} else if current != nil {
+		return false
+	}
+
+	return true
+}
+
+type updater struct {
+	client kafkaiface.KafkaAPI
+}
+
+func (u *updater) update(ctx context.Context, mg cpresource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+	cr, ok := mg.(*svcapitypes.Cluster)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+	if aws.StringValue(cr.Status.AtProvider.State) != stateActive {
+		return managed.ExternalUpdate{}, errors.New(errStateForUpdate)
+	}
+	input := GenerateDescribeClusterInput(cr)
+
+	if cr.Status.AtProvider.ClusterARN != nil {
```

Comment:
`meta.GetExternalName(cr)` should be used to get the cluster ARN.

Reply:
Thanks for your suggestions!